### PR TITLE
[FIX] website: not fail when no dynamic filter is defined

### PR DESCRIPTION
**Pull request by bso-odoo**
Since [1] when only website is installed, dropping a dynamic snippet
produces an error because no dynamic filter is defined and it tries to
specify one selected by default.

This commit avoids setting a default filter when there is none.

It did work in 15.0 because the log was inside
`_renderDynamicFiltersSelector` which contained an
`if (dynamicFilters.length > 0)` that avoided the problem.
In [2], when most of that logic was moved from the render method to the
fetch method, that condition was lost.

Steps to reproduce:
- Install website only
- Go to debug mode
- Drop a "Dynamic Snippet" or a "Dynamic Carousel"
=> Traceback appears because there cannot be a default filter when there
is no filter.

[1]: https://github.com/odoo/odoo/commit/3355dc16235355fe51e894f14e275210464608c6
[2]: https://github.com/odoo/odoo/commit/9e0b398fe608e716835260c5f9923f933d723f8e

opw-3166634

### CODE REVIEW

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
I tested in 15.0, I don't have a crash but I got this:
image

1. Should not that be fixed too? (I guess it's the same in 16.0 with the error fixed?)
2. Why is there no crash in 15.0?

**Comment by bso-odoo** [comment]
> Why is there no crash in 15.0 ?

In 15.0, there was a dedicated `_renderDynamicFiltersSelector` which contained an `if (dynamicFilters.length > 0)` that avoided the problem.
In https://github.com/odoo/odoo/commit/9e0b398fe608e716835260c5f9923f933d723f8e, when most of that logic was moved from the render method to the fetch method, that condition was lost.

I have now included this information in the commit message.

**Comment by bso-odoo** [comment]
@qsm-odoo Regarding the flat empty dropdowns, I made a second commit that adds (and ignores) a fake item. (Otherwise, the we-select is not even involved at all in the `updateUI`.)
Tell me if you want me to do it in an earlier version instead.

File: addons/website/static/src/snippets/s_dynamic_snippet/options.js [hunk 1]
```diff
@@ -225,6 +232,13 @@ const dynamicSnippetOptions = options.Class.extend({
             }
             selectUserValueWidgetElement.appendChild(button);
         }
+        if (!Object.keys(data).length) {
+            const button = document.createElement('we-button');
+            button.dataset.selectDataAttribute = 'empty';
+            button.dataset.emptyListPlaceholder = true;
+            button.innerText = '/';
+            selectUserValueWidgetElement.appendChild(button);
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_dynamic_snippet/options.js:240]: The problem is the same in 14.0/15.0, no? So yes, I would make another PR for this commit, targeting 14.0 :+1: 

I would fix this differently:
- Not link it to the option/selectDataAttribute, this should be a feature at the UserValueWidget level
- Can't we have a pure CSS solution? (something like we-select:empty::before { content: "/"; .... };

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

**Comment by bso-odoo** [comment]
@qsm-odoo Rebased on top of the now merged #112117.
Re-tested locally and runbot is green. I believe this can now be merged too.

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by bso-odoo** [comment]
@robodoo retry

**Comment by qsm-odoo** [comment]
@robodoo retry

Let's retry one more time. Afterwards, I guess we'll try to rebase and r+ again but I don't see what the difference would be :shrug: This was green, r+, staging fails because of a mass mailing error (not related to this) so retry should make it succeed as I see other PR succeed...
